src/components/DateTimePicker/TimePicker.tsx

```diff
@@ -52,22 +58,93 @@ export class TimePicker extends React.Component<ITimePickerProps, void> {
         this.props.onTimeChanged(this.props.hour, value);
     }
 
+    normalizeHours(hour: number, is24Hour: boolean, isPeriodPm: boolean) {
+        if (!is24Hour) {
+            if (hour > 12) {
+                hour = 12;
+            }
+            if (isPeriodPm) {
+                if (hour < 12) {
+                    hour += 12;
+                }
+            } else {
+                if (hour === 12) {
+                    hour = 0;

```

Comment:
i think that the problem is that i am not quite sure if this function return the display value [1-12] or [0-24], the actual value. The **if hour > 12** line leads me to believe that we are working within the [1-12] range as an output, but for the AM part of the time it sets 0 which can never be. So i'm a bit confused. the this.props.OnTimeChangeCall should work with [0-23] since we do not know if it is AM or PM, but from the code, it will be clipped to 12 hours.